### ISSUE TITLE
Positive lookahead don't work with range

### ISSUE THREAD

**Issue by 1aam2am1** [hanickadot/compile-time-regular-expressions#191]
I have a regex: "(?=((?:55|66)[0-9a-fA-F]{8,}\r))".
That will for this input: "550020660232020202125\r",  And std::sregex_iterator will return 2 matches:

- 550020660232020202125\r
- 660232020202125\r

But ctre::range will only return "" and will never end.

**Comment by hanickadot** [comment]
Hi, thanks for reporting this issue. CTRE is not returning any result in the implicit "zero" capture, because the positive look-ahead is not consuming anything. If you want to access captured content, use `.get<1>()` from result.

About the infinite loop, I will look at it. meanwhile if you add `.` at the end of your regex you will avoid it.

**Comment by 1aam2am1** [comment]
Yes, I have forgotten to copy get<1>() when i was testing if regex is faster or my own function. With . at the end it works.
